plugins/s3_auth/s3_auth.cc

```diff
@@ -440,6 +594,55 @@ str_concat(char *dst, size_t dst_len, const char *src, size_t src_len)
   return to_copy;
 }
 
+TSHttpStatus
+S3Request::authorize(S3Config *s3)
+{
+  TSHttpStatus status = TS_HTTP_STATUS_INTERNAL_SERVER_ERROR;
+  switch (s3->version()) {
```

Comment:
Maybe we should have a proper enum for the supported versions ? Not particularly important, so either way is ok with me.